modules/xmpp/SdpConsistency.js

```diff
@@ -0,0 +1,176 @@
+import * as transform from 'sdp-transform';
+
+/**
+ * Begin helper functions
+ */
+/**
+ * Given a video mline (as parsed from transform.parse),
+ *  return the single primary video ssrcs
+ * @param {object} videoMLine the video MLine from which to extract the
+ *  primary video ssrc
+ * @returns {number} the primary video ssrc
+ */
+function getPrimarySsrc (videoMLine) {
+    if (!videoMLine.ssrcs) {
+        return;
+    }
+    let numSsrcs = videoMLine.ssrcs
+        .filter(ssrcInfo => ssrcInfo.attribute === "msid")
+        .map(ssrcInfo => ssrcInfo.id)
+        .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
+        .length;
+    if (numSsrcs === 1) {
+        return videoMLine.ssrcs
+            .filter(ssrcInfo => ssrcInfo.attribute === "msid")
+            .map(ssrcInfo => ssrcInfo.id)[0];
+    } else {
+        let findGroup = (mLine, groupName) => {
+            return mLine
+                .ssrcGroups
+                .filter(group => group.semantics === groupName)[0];
+        };
+        // Look for a SIM or FID group
+        if (videoMLine.ssrcGroups) {
+            let simGroup = findGroup(videoMLine, "SIM");
+            if (simGroup) {
+                return parseInt(simGroup.ssrcs.split(" ")[0]);
+            }
+            let fidGroup = findGroup(videoMLine, "FID");
+            if (fidGroup) {
+                return parseInt(fidGroup.ssrcs.split(" ")[0]);
+            }
+        }
+    }
+}
+
+/**
+ * Given a video mline (as parsed from transform.parse),
+ *  and a primary ssrc, return the corresponding rtx ssrc
+ *  (if there is one) for that video ssrc
+ * @param {object} videoMLine the video MLine from which to extract the
+ *  rtx video ssrc
+ * @param {number} primarySsrc the video ssrc for which to find the
+ *  corresponding rtx ssrc
+ * @returns {number} the rtx ssrc (or undefined if there isn't one)
+ */
+function getRtxSsrc (videoMLine, primarySsrc) {
+    if (videoMLine.ssrcGroups) {
+        return videoMLine
```

Comment:
i don't find the above to be very easy to read (primarily because of the long line with the call to parseInt inside the if), and i think assigning to an outside variable inside the 'find' doesn't feel great, but i did change this to use find to find the specific group and then move the rtxSsrc parsing (once the group has been found) outside so the .map/.filter chain is gone.